presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/DetermineJoinDistributionType.java

```diff
@@ -148,21 +146,33 @@ private JoinNode getSizeBasedJoin(JoinNode joinNode, Context context)
         // the output from a filter or aggregation due to lack of estimates.
         // We use getFirstKnownOutputSizeInBytes instead of getSourceTablesSizeInBytes to account for the reduction in
         // output size from the operators between the join and the table scan as much as possible when comparing the sizes of the join sides.
-        double leftOutputSizeInBytes = getFirstKnownOutputSizeInBytes(joinNode.getLeft(), context);
-        double rightOutputSizeInBytes = getFirstKnownOutputSizeInBytes(joinNode.getRight(), context);
+
         // All the REPLICATED cases were handled in the code above, so now we only consider PARTITIONED cases here
-        if (rightOutputSizeInBytes * SIZE_DIFFERENCE_THRESHOLD < leftOutputSizeInBytes && !mustReplicate(joinNode, context)) {
+        if (isMuchSmaller(joinNode.getRight(), joinNode.getLeft(), context) && !mustReplicate(joinNode, context)) {
             return joinNode.withDistributionType(PARTITIONED);
         }
 
-        if (leftOutputSizeInBytes * SIZE_DIFFERENCE_THRESHOLD < rightOutputSizeInBytes && !mustReplicate(flippedJoin, context)) {
+        if (isMuchSmaller(joinNode.getLeft(), joinNode.getRight(), context) && !mustReplicate(flippedJoin, context)) {
             return flippedJoin.withDistributionType(PARTITIONED);
         }
 
         // neither side is small enough, choose syntactic join order
         return getSyntacticOrderJoin(joinNode, context, AUTOMATIC);
     }
 
+    public static boolean isSmall(PlanNode planNode, Context context)
```

Comment:
"small" is too generic. Function name should say that we are checking small enough for broadcast. Like "fitsBroadcastSize" or something